clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -175,6 +176,9 @@ public double compressionRate() {
      * @return The built log buffer
      */
     public MemoryRecords build() {
+        if (aborted) {
```

Comment:
We should add this check in `ensureOpenForRecordBatchWrite`,  `ensureOpenForRecordAppend`, and `close` as well.